admin-dev/themes/new-theme/js/pages/product/components/combination-list/bulk-delete-handler.ts

```diff
@@ -100,49 +98,65 @@ export default class BulkDeleteHandler {
     });
   }
 
-  /**
-   * @todo: the bulk delete action should be displayed with a progress modal once it is ready.
-   */
   private async bulkDelete(combinationIds: number[]): Promise<void> {
-    const progressModal = this.showProgressModal();
-    const progressModalElement = document.getElementById(CombinationMap.bulkProgressModalId);
+    const $bulkDeleteBtn = $(CombinationMap.bulkDeleteBtn);
+    const bulkChunkSize = Number($bulkDeleteBtn.data('bulkChunkSize'));
+    const abortController = new AbortController();
+
+    const progressModal = new ProgressModal({
+      id: CombinationMap.bulkProgressModalId,
+      abortCallback: () => {
+        stopProcess = true;
+        abortController.abort();
+      },
+      closeCallback: () => this.eventEmitter.emit(CombinationEvents.bulkDeleteFinished),
+      progressionTitle: $bulkDeleteBtn.data('progressTitle'),
+      progressionMessage: $bulkDeleteBtn.data('progressMessage'),
+      closeLabel: $bulkDeleteBtn.data('closeLabel'),
+      abortProcessingLabel: $bulkDeleteBtn.data('stopProcessing'),
+      errorsMessage: $bulkDeleteBtn.data('errorsMessage'),
+      backToProcessingLabel: $bulkDeleteBtn.data('backToProcessing'),
+      downloadErrorLogLabel: $bulkDeleteBtn.data('downloadErrorLog'),
+      viewErrorLogLabel: $bulkDeleteBtn.data('viewErrorLog'),
+      viewErrorTitle: $bulkDeleteBtn.data('viewErrorTitle'),
+      total: combinationIds.length,
+    });
+    progressModal.show();
+    let stopProcess = false;
+    let doneCount = 0;
+    while (combinationIds.length) {
+      const chunkIds: number[] = combinationIds.splice(0, bulkChunkSize);
+
+      if (stopProcess) {
+        break;
+      }
 
-    let progress = 1;
+      let data: Record<string, any>;
 
-    //@todo: is there better loop option to avoid disabling eslint?
-    // eslint-disable-next-line no-restricted-syntax
-    for (const combinationId of combinationIds) {
       try {
         // eslint-disable-next-line no-await-in-loop
-        await this.combinationsService.deleteCombination(combinationId);
-      } catch (error) {
-        console.error(error);
+        data = await this.combinationsService.bulkDeleteCombinations(this.productId, chunkIds);
+      } catch (e) {
+        data = {
+          errors: e.responseJSON.errors ?? null,
+          error: `Something went wrong with IDs ${chunkIds.join(', ')}: ${e.message ?? ''}`,
+        };
       }
 
-      //@todo: also related with temporary progress modal. Needs to be fixed according to new progress modal once its merged in #26004.
-      const progressContent = progressModalElement?.querySelector<HTMLParagraphElement>('.progress-increment');
-
-      if (progressContent) {
-        progressContent.innerHTML = String(progress);
+      doneCount += chunkIds.length;
+      progressModal.updateProgress(doneCount);
+
+      if (!data.success) {
+        if (data.errors && Array.isArray(data.errors)) {
+          data.errors.forEach((error:string) => {
```

Comment:
```suggestion
          data.errors.forEach((error: string) => {
```

nitpick but global styling